Bottom: aa9f11498e1633caf7c86e13d32d67ebea1d2248
Top:    1c48b13a7f6f0a8f24ba7ab90c4ea9d9a49ebffc
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-07 23:31:43 +0100

Refresh of move-the-selectpin-function-in

---

diff --git a/src/hid/gtk/gui-netlist-window.c b/src/hid/gtk/gui-netlist-window.c
index 8598018..ea7cef5 100644
--- a/src/hid/gtk/gui-netlist-window.c
+++ b/src/hid/gtk/gui-netlist-window.c
@@ -207,6 +207,30 @@ node_model_update (LibraryMenuType * menu)
     g_object_unref (G_OBJECT (model));
 }
 
+static void
+toggle_pin_selected (LibraryEntryType *entry)
+{
+  ConnectionType conn;
+
+  if (!SeekPad (entry, &conn, false))
+    return;
+
+  AddObjectToFlagUndoList (conn.type, conn.ptr1, conn.ptr2, conn.ptr2);
+  TOGGLE_FLAG (SELECTEDFLAG, (AnyObjectType *)conn.ptr2);
+  DrawObject (conn.type, conn.ptr1, conn.ptr2);
+
+  if (conn.type == PIN_TYPE)
+    {
+      PinTypePtr pin = (PinTypePtr) conn.ptr2;
+      CenterDisplay (pin->X, pin->Y);
+    }
+  else if (conn.type == PAD_TYPE)
+    {
+      PadTypePtr pad = (PadTypePtr) conn.ptr2;
+      CenterDisplay (pad->Point1.X, pad->Point1.Y);
+    }
+}
+
 
 /* Callback when the user clicks on a PCB node in the right node treeview.
  */
@@ -234,7 +258,7 @@ node_selection_changed_cb (GtkTreeSelection * selection, gpointer data)
       |  if off here will get our on/off toggling out of sync.
       */
       if (node_net == node_selected_net)
-	TogglePinSelected (node);
+        toggle_pin_selected (node);
       g_free (node_name);
       node_name = NULL;
     }
@@ -259,7 +283,7 @@ node_selection_changed_cb (GtkTreeSelection * selection, gpointer data)
 
   /* Now just toggle a select of the node on the layout and pan.
    */
-  TogglePinSelected (node);
+  toggle_pin_selected (node);
   IncrementUndoSerialNumber ();
 	margin = gport->view_width / 20;
 	if (   Crosshair.X < gport->view_x0 + margin
diff --git a/src/select.c b/src/select.c
index 99fca0a..96db234 100644
--- a/src/select.c
+++ b/src/select.c
@@ -61,47 +61,6 @@
 
 RCSID ("$Id$");
 
-
-
-/* ---------------------------------------------------------------------------
- * toggle selection of pin
- */
-void
-TogglePinSelected (LibraryEntryTypePtr entry)
-{
-  ConnectionType conn;
-
-  if (SeekPad (entry, &conn, false))
-    {
-      switch (conn.type)
-	{
-	case PIN_TYPE:
-	  {
-	    PinTypePtr pin = (PinTypePtr) conn.ptr2;
-
-	    AddObjectToFlagUndoList (PIN_TYPE, conn.ptr1, conn.ptr2,
-				     conn.ptr2);
-	    TOGGLE_FLAG (SELECTEDFLAG, pin);
-	    CenterDisplay (pin->X, pin->Y);
-	    DrawPin (pin);
-	    break;
-	  }
-	case PAD_TYPE:
-	  {
-	    PadTypePtr pad = (PadTypePtr) conn.ptr2;
-
-	    AddObjectToFlagUndoList (PAD_TYPE, conn.ptr1, conn.ptr2,
-				     conn.ptr2);
-	    TOGGLE_FLAG (SELECTEDFLAG, pad);
-	    CenterDisplay (pad->Point1.X, pad->Point1.Y);
-	    DrawPad (pad);
-	    break;
-	  }
-	}
-    }
-}
-
-
 /* ---------------------------------------------------------------------------
  * toggles the selection of any kind of object
  * the different types are defined by search.h
diff --git a/src/select.h b/src/select.h
index 513be4b..0e12cc2 100644
--- a/src/select.h
+++ b/src/select.h
@@ -37,7 +37,6 @@
 	(VIA_TYPE | LINE_TYPE | TEXT_TYPE | POLYGON_TYPE | ELEMENT_TYPE |	\
 	 PIN_TYPE | PAD_TYPE | ELEMENTNAME_TYPE | RATLINE_TYPE | ARC_TYPE)
 
-void TogglePinSelected (LibraryEntryTypePtr entry);
 bool SelectObject (void);
 bool SelectBlock (BoxTypePtr, bool);
 bool SelectedOperation (ObjectFunctionTypePtr, bool, int);
